core/arch/x86_64/plat-kbl/main.c

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Intel Corporation
+ * Copyright (c) 2016, Linaro Limited
+ * Copyright (c) 2014, STMicroelectronics International N.V.
+ */
+
+#include <console.h>
+#include <drivers/gic.h>
+#include <drivers/pl011.h>
+#include <drivers/uart.h>
+#include <drivers/tzc400.h>
+#include <initcall.h>
+#include <keep.h>
+#include <kernel/boot.h>
+#include <kernel/misc.h>
+#include <kernel/panic.h>
+#include <kernel/tee_time.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <stdint.h>
+#include <string.h>
+#include <tee/entry_fast.h>
+#include <tee/entry_std.h>
+#include <trace.h>
+
+extern uint32_t is_optee_boot_complete;
+
+static struct uart_data console_data __nex_bss;
+
+#if defined(PLATFORM_FLAVOR_fvp)
+register_phys_mem(MEM_AREA_RAM_SEC, TZCDRAM_BASE, TZCDRAM_SIZE);
+#endif
+
+register_phys_mem(MEM_AREA_IO_SEC, CONSOLE_UART_BASE1, PL011_REG_SIZE);
+register_phys_mem(MEM_AREA_IO_SEC, CONSOLE_UART_BASE2, PL011_REG_SIZE);
+register_phys_mem(MEM_AREA_IO_SEC, APIC_BASE, APIC_REG_SIZE);
+
+void console_init(void)
+{
+	if (is_optee_boot_complete == 0) {
+		uart_init(&console_data, CONSOLE_UART_BASE1);
```

Comment:
Since you only have one serial address, just need to do uart_init once with CONSOLE_UART_BASE1.

Reply:
Please remove if (is_optee_boot_complete == 0) this condition check.